C702.tex

```diff
@@ -0,0 +1,160 @@
+The name attribute is required and corresponds to the name of the object in the target database.
+The description is an optional, human-readable description of the object.
+The identifier provides a way to uniquely identify the object within the schema and is primarily used for object referencing.
+Identifiers may be generated automatically or provided explicitly by the user.
+
+A version attribute indicates the version of the schema and can be used to track changes over time.
+In addition to the version string, lists of compatible and read-compatible versions can also be provided.
+Felis does not enforce any particular versioning scheme, but semantic versioning is recommended.
+
+A schema must define at least one table, and each table must include a list of one or more columns.
+Columns must have a \texttt{datatype} field and may include a number of optional properties which are used for a variety of purposes, including specifying semantics of the column, such as units, UCD, and xtype.
+These properties can also specify the column's behavior in a database, such as whether it is nullable.
+Both tables and columns may define a TAP index, typically used to recommend an ordering for clients.
+
+Constraints define rules that restrict column values in a table.
+Felis supports primary key, foreign key, and unique constraints.
+Each constraint defines the columns to which it applies, as well as the target primary key for a foreign key constraint.
+
+Indexes indicate that a column or set of columns should be indexed in the database for faster query performance.
+Indexes are defined by a list of one or more columns.
+The indexes may be used as an indication to clients that they should consider using these columns in their queries for better performance.
+
+\section{Python Data Model}
+
+\begin{description}
+    \item[Schema Validation]
+    The YAML data defining the schema is validated when loaded, ensuring that the schema is correctly defined.
+    The model comprises a set of Python classes which inherit from Pydantic's \texttt{BaseModel} class and correspond to the different types of schema objects.
+    The attributes of the classes are defined using Pydantic's \texttt{Field} class, which allows for specifying the data type, default value, and validation rules.
+    This allows for strict validation of the schema, ensuring that all required fields are present and that the data types are correct.
+
+    Additionally, various "business rules" are defined and enforced using Python "validator" functions.
+    These functions run automatically as part of the validation process.
+    An example business rule might ensure that table names are unique within a given schema.
+    Any validation errors which occur will cause the validation process to fail, preventing the schema from being loaded, and an error message will be returned to the user.
+\end{description}
+
+\articlefigure{C702_f2.eps}{fig:python}{Example of using the Felis Python API.}
```

Comment:
Could the code example be rendered as an inline text, I'm not sure that having it as an image helps the representation? (And dark background wastes ink in case someone decides to print it).

Reply:
Good suggestion - I'll go ahead and do this. It will free up some lines for additional text, too. :)